### PR TITLE
[ADR] Changing username

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: src/adr/0065-changing-username.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- `ACLUsernameChangeTaskStep`: migrates ACLs on mailboxes the migrated user has access to and updates subscriptions accordingly.
+
+We introduce `fromStep` query parameter that allows skipping previous steps, allowing to resume the username change from a failed step.
+This option could ease operators in case the data migration fails in the middle.
+
+## Consequences
+- Add one more useful feature to James rich feature set: Users can change their name while preserving their email data.
+- User would be aware that they have a new account through new username/password. No implicit migration without user attention is archived today.
+- JMAP Identity data is not migrated because they are tightly coupled to old username, user should create identities again themselves.
+- Some data considered not critical like Push subscription, Vacation response,... are not migrated today. Contributions for them is welcome.
```
Comment (quantranhong1999) [inline src/adr/0065-changing-username.md:48]: BTW are we missing migration for Quota?

Comment (chibenwa) [reply src/adr/0065-changing-username.md]: Very good question!

Ticket?

File: src/adr/0065-changing-username.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Date: 2023-03-17
+
+## Status
+
+Accepted (lazy consensus).
+
+Implemented.
+
+## Context
+Changing username is a desired feature for an email server. It is a common practice when a user marries. For example:
```
Comment (chibenwa) [inline src/adr/0065-changing-username.md:12]: Wedding + divorce are only exemple

File: src/adr/0065-changing-username.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Alice MARTINEZ have an email address `alice.martinez@domain.tld` and gets married to Bob ERNANDEZ. She thus wants to
+change her email address to `alice.ernandez@domain.tld` while preserving her email data.
+
+Nowadays, James uses username as an identifier for user data. Therefore, this feature is hard to implement because of changing
+directly identifier to another value is difficult and could break data consistency if we do not do it carefully.
+
+## Decision
+
+Need to copy the data
+Modular design
```
Comment (chibenwa) [inline src/adr/0065-changing-username.md:23]: Not redacted well?

File: src/adr/0065-changing-username.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+We introduce `fromStep` query parameter that allows skipping previous steps, allowing the operator to resume the username change from a failed step.
+This option could ease operators in case the data migration fails in the middle.
+
+## Consequences
+- Add one more useful feature to James rich feature set: Users can change their name while preserving their email data.
+- Users would be aware that they have a new account through a new username/password. No implicit migration without user attention is archived today.
+- JMAP Identity data is not migrated because they are tightly coupled to the old username, users should create identities again themselves.
+- Some data considered not critical like Push subscription, Vacation response,... are not migrated today. Contributions for them are welcome.
+
```
Comment (chibenwa) [inline src/adr/0065-changing-username.md:49]: Consequence: changing usernames adapts to each james backend and is extension friendly?

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**